### ISSUE TITLE
Segment timestamps are buggy in BatchedInferencePipeline

### ISSUE THREAD

**Issue by kalradivyanshu** [SYSTRAN/faster-whisper#919]
In the recent code #856, `BatchedInferencePipeline` the segments are sometimes wrong for some reason.

I recorded an audio clip: https://drive.google.com/file/d/1cbDbiXi12SIsd0hIDfs61VdtgI78Fg_p/view?usp=sharing

and for this `BatchedInferencePipeline` gives `Segment(id=1, seek=2307, start=18.71, end=23.07...` even though the clip is only 3 second long.

Here is a colab recreating this issue, just upload batch.wav from the link above: https://colab.research.google.com/drive/1ie7uMFW_LJUvxGHW3KkT5iG8uUZiTVwU?usp=sharing

```py
segments, info = batched_model.transcribe(arr, word_timestamps=True, batch_size = 1)
print(list(segments))
```
output:
```
[Segment(id=1, seek=2307, start=18.71, end=23.07, text=' Hey Michael, how are you?', tokens=[14690, 3899, 11, 703, 389, 345, 30], avg_logprob=-0.21215820871293545, compression_ratio=0.7647058823529411, no_speech_prob=0.08087158203125, words=[Word(start=18.71, end=20.11, word=' Hey', probability=0.71240234375), Word(start=20.11, end=20.11, word=' Michael,', probability=0.732421875), Word(start=22.71, end=22.71, word=' how', probability=0.98046875), Word(start=22.71, end=22.71, word=' are', probability=0.9990234375), Word(start=22.71, end=23.07, word=' you?', probability=0.99853515625)], temperature=1.0)]
```

```py
segments, info = model.transcribe(arr, word_timestamps=True)
print(list(segments))
```
correct output:
```
[Segment(id=1, seek=330, start=0.9999999999999996, end=2.5, text=' Hey Michael, how are you?', tokens=[50413, 14690, 3899, 11, 703, 389, 345, 30, 50513], avg_logprob=-0.568749976158142, compression_ratio=0.7575757575757576, no_speech_prob=0.078125, words=[Word(start=0.9999999999999996, end=1.4, word=' Hey', probability=0.7177734375), Word(start=1.4, end=1.6, word=' Michael,', probability=0.68896484375), Word(start=2.04, end=2.14, word=' how', probability=0.98046875), Word(start=2.14, end=2.32, word=' are', probability=0.998046875), Word(start=2.32, end=2.5, word=' you?', probability=0.99755859375)], temperature=0.0)]
```

Thank you for all your work!

**Comment by Jiltseb** [comment]
@MahmoudAshraf97

**Comment by MahmoudAshraf97** [comment]
should be fixed in #920

**Comment by MahmoudAshraf97** [comment]
because the input is not the same
the `end` of a vad segment should never be more than the end of its last subsegment, in your case the end is 30.0 while the subsegment end is 7.024
this is equivalent to 
`clip_timestamps = [0.0, 30.0]`
